transmission_interface/CMakeLists.txt

```
@@ -12,23 +12,24 @@ endif()
 
 # find dependencies
 find_package(ament_cmake REQUIRED)
-find_package(hardware_interface REQUIRED)
-find_package(tinyxml2_vendor REQUIRED)
-find_package(TinyXML2 REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(hardware_interface REQUIRED)    # todo: we should be using this for hardware interface constants
+find_package(tinyxml_vendor REQUIRED)
+find_package(TinyXML REQUIRED)  # provided by tinyxml2 upstream, or tinyxml2_vendor
```

```suggestion
find_package(tinyxml2_vendor REQUIRED)
find_package(TinyXML2 REQUIRED)  # provided by tinyxml2 upstream, or tinyxml2_vendor
```
we are using this version [here](https://github.com/ros-controls/ros2_control/blob/master/hardware_interface/CMakeLists.txt)

---

Upgraded to TinyXML2